datafusion/sql/src/expr/value.rs

```diff
@@ -191,6 +200,76 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             SQLExpr::Value(
                 Value::SingleQuotedString(s) | Value::DoubleQuotedString(s),
             ) => s,
+            SQLExpr::BinaryOp { left, op, right } => {
```

Comment:
Your PR description is beautiful -- I thin some part if it would be valuable to include in the comments as well for future readers:

```suggestion
            // Support expressions like `interval '1 month' + date/timestamp`. 
            // Such expressions are parsed like this by sqlparser-rs 
            // 
            // Interval
            // BinaryOp
            //   Value(StringLiteral)
            //   Cast
            //     Value(StringLiteral)
            // 
            // This code rewrites them to the following:
            //
            // BinaryOp
            //   Interval
            //     Value(StringLiteral)
            //   Cast
            //      Value(StringLiteral)
            SQLExpr::BinaryOp { left, op, right } => {
```

Reply:
Thank you